File: packages/py/src/airlock/application/gate_policy.py
"""The default gate policy."""

from airlock.application.ports.gate_policy import GateDecisionInput
from airlock.domain.tool import RiskTier


class RiskBasedGatePolicy:
    """Tools tagged `sensitive` require human approval; `safe` tools run
    automatically (see ADR-0003)."""

    def requires_approval(self, decision_input: GateDecisionInput) -> bool:
        return decision_input.tool.risk == RiskTier.SENSITIVE

File: packages/ts/src/application/gate-policy.ts
import { RiskTier } from "../domain/tool";
import type { GateDecisionInput, GatePolicy } from "./ports/gate-policy";

/**
 * Default gate policy: tools tagged `sensitive` require human approval; `safe`
 * tools run automatically (see ADR-0003). The rule lives here, not in the tool
 * definitions or the prompt, so it is deterministic and testable.
 */
export class RiskBasedGatePolicy implements GatePolicy {
  requiresApproval(input: GateDecisionInput): boolean {
    return input.tool.risk === RiskTier.Sensitive;
  }
}
